### PR TITLE
backport: Merge bitcoin#18732,18859,18088,18665,18733,18756,18754,18777,18437,18011

### PR DESCRIPTION
**Pull request by vijaydasmp**
## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -114,6 +114,48 @@ v18.2.1 Change log
 
 See detailed [set of changes](https://github.com/dashpay/dash/compare/v18.2.0...dashpay:v18.2.1).
 
+P2P and network changes
```
Comment (UdjinM6) [inline doc/release-notes.md:117]: 18733: this whole text should go into a separate `release-notes-18733.md` file

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by knst — DISMISSED** [review]

File: test/functional/test_framework/script_util.py [hunk 2]
```diff
@@ -23,3 +23,4 @@
 # scriptPubKeys are needed, to guarantee that the minimum transaction size is
 # met.
 DUMMY_P2SH_SCRIPT = CScript([b'a' * 22])
+DUMMY_2_P2WPKH_SCRIPT = CScript([b'b' * 22])
```
[the remark this answers is not in the document]
Comment (vijaydasmp) [reply test/functional/test_framework/script_util.py:26]: ok

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 @knst @PastaPastaPasta , please review

**Review by knst — APPROVED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 @PastaPastaPasta , please review

**Comment by vijaydasmp** [comment]
Dear @UdjinM6 @PastaPastaPasta please review

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**